src/explore-education-statistics-admin/src/pages/release/pre-release/components/PreReleaseUserAccessForm.tsx

```diff
@@ -51,18 +59,31 @@ const PreReleaseUserAccessForm = ({
     releaseId,
   ]);
 
-  const handleSubmit = useFormSubmit<FormValues>(async (values, actions) => {
-    const newUser = await preReleaseUserService.inviteUser(
-      releaseId,
-      values.email,
+  const [invitePlan, setInvitePlan] = useState<PreReleaseInvitePlan>();
+
+  const handleSubmit = useFormSubmit<FormValues>(async values => {
+    setInvitePlan(
+      await preReleaseUserService.getInvitePlan(releaseId, values.emails),
     );
+  }, errorMappings);
 
-    setUsers({
-      value: [...users, newUser],
-    });
+  const handleModalSubmit = useFormSubmit<FormValues>(
+    async (values, actions) => {
+      await preReleaseUserService
+        .inviteUsers(releaseId, values.emails)
+        .then(newUsers => {
```

Comment:
No need to use `then` as we're in an async/await block. Can be rewritten as:

```ts
const newUsers = await preReleaseUserService.inviteUsers(releaseId, values.emails);

setUsers({
  value: [...users, ...newUsers],
});

actions.resetForm();
```

Reply:
That's good to know. Thanks for the tip.